mantidproject/mantid
# Bump Eigen to 3.4.0

**Description of work.**
Bumps Eigen to the 3.4 series. We could stick with 3.3.9 to fix
deprecated C++-17 warnings, but as there is only one breaking change
with 3.4.0 lets go all the way up in one single hit

**To test:**
- Check that all tests pass

There is no associated issue. - Fixes an issue seen on CI with newer MSVC compiler

*This does not require release notes* because it's not a user visible change



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
